### PR TITLE
[compiler v2] Fixing an issue with common namespace for all scripts being compiled.

### LINKED ISSUE #11341: [Bug][Compiler-v2] v2 complains duplicated declaration in two scripts
# 🐛 Bug

When compiling the [upgrade](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/upgrade_and_govern/upgrade) module, the V2 will generate the following error:

```
error: duplicate declaration of `PARAMETER_1`
  ┌─ /Users/tengzhang/aptos-core/aptos-move/move-examples/upgrade_and_govern/upgrade/scripts/set_only.move:5:11
  │
5 │     const PARAMETER_1: u64 = 500;
  │           ^^^^^^^^^^^

error: duplicate declaration of `PARAMETER_2`
  ┌─ /Users/tengzhang/aptos-core/aptos-move/move-examples/upgrade_and_govern/upgrade/scripts/set_only.move:6:11
  │
6 │     const PARAMETER_2: u64 = 700;

```

### PR DESCRIPTION
**Pull request by vineethk**
### Description

Closes #11341.

Because all the scripts being compiled used the same namespace (`<SELF>`), it could result in conflicts when same names were used for entities. We now instead give each script being compiled a unique namespace. Also contains minor cleanups along the way.

### Test Plan

Manually tested compiling the failing module mentioned in the bug: it now compiles successfully.

### CODE REVIEW

**Review by brmataptos — APPROVED** [review]
Does this also work for transactional tests?  Perhaps the driver needs modifications, but having something automated that uses multiple scripts might be a good idea.

**Comment by vineethk** [comment]
> Does this also work for transactional tests? Perhaps the driver needs modifications, but having something automated that uses multiple scripts might be a good idea.

@brmataptos As far as I can tell, you can only specify one script per action in a transactional test (otherwise, you get an error). If you have multiple actions, then they are all getting compiled separately, so they don't really have clashing namespaces.

**Review by rahxephon89 — APPROVED** [review]

File: third_party/move/move-core/types/src/account_address.rs [hunk 1]
```diff
@@ -18,6 +18,8 @@ impl AccountAddress {
     /// The number of bytes in an address.
     /// Default to 16 bytes, can be set to 20 bytes with address20 feature.
     pub const LENGTH: usize = 32;
+    /// Max address: 0xff....
+    pub const MAX_ADDRESS: Self = Self([0xFF; Self::LENGTH]);
```
Comment (wrwg) [inline third_party/move/move-core/types/src/account_address.rs:22]: Does this notation actually fill up FF for all 32 bytes? If so, never stop learning about Rust

File: third_party/move/move-core/types/src/identifier.rs [hunk 2]
```diff
@@ -57,10 +57,24 @@ const fn all_bytes_valid(b: &[u8], start_offset: usize) -> bool {
...
+    }
+    true
+}
+
 /// Describes what identifiers are allowed.
 ///
 /// For now this is deliberately restrictive -- we would like to evolve this in the future.
-// TODO: "<SELF>" is coded as an exception. It should be removed once CompiledScript goes away.
+/// TODO: "<SELF>" and "<SELF>_[0-9]+" are coded as exceptions.
+///       They should be removed once CompiledScript goes away.
```
Comment (wrwg) [inline third_party/move/move-core/types/src/identifier.rs:77]: CompiledScript will likely not ever go away.

File: third_party/move/move-core/types/src/account_address.rs (hunk 1 above)
Comment (vineethk) [reply third_party/move/move-core/types/src/account_address.rs:22]: Yes. Note that I used the same notation as was there before: I just changed the hard coded usages to instead use a public constant.

**Pull request merged**